lxd/network_load_balancers.go

```diff
@@ -0,0 +1,548 @@
+// Returns a list of network address load balancers (structs).
+//
+// ---
+// produces:
+//   - application/json
+// parameters:
+//   - in: query
+//     name: project
+//     description: Project name
+//     type: string
+//     example: default
+// responses:
+//   "200":
+//     description: API endpoints
+//     schema:
+//       type: object
+//       description: Sync response
+//       properties:
+//         type:
+//           type: string
+//           description: Response type
+//           example: sync
+//         status:
+//           type: string
+//           description: Status description
+//           example: Success
+//         status_code:
+//           type: integer
+//           description: Status code
+//           example: 200
+//         metadata:
+//           type: array
+//           description: List of network address load balancers
+//           items:
+//             $ref: "#/definitions/NetworkLoadBalancer"
+//   "403":
+//     $ref: "#/responses/Forbidden"
+//   "500":
+//     $ref: "#/responses/InternalServerError"
+func networkLoadBalancersGet(d *Daemon, r *http.Request) response.Response {
+	projectName, _, err := project.NetworkProject(d.State().DB.Cluster, projectParam(r))
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	networkName, err := url.PathUnescape(mux.Vars(r)["networkName"])
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	n, err := network.LoadByName(d.State(), projectName, networkName)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("Failed loading network: %w", err))
+	}
+
+	if !n.Info().LoadBalancers {
+		return response.BadRequest(fmt.Errorf("Network driver %q does not support load balancers", n.Type()))
+	}
+
+	memberSpecific := false // Get load balancers for all cluster members.
+
+	if util.IsRecursionRequest(r) {
+		records, err := d.State().DB.Cluster.GetNetworkLoadBalancers(n.ID(), memberSpecific)
+		if err != nil {
+			return response.SmartError(fmt.Errorf("Failed loading network load balancers: %w", err))
+		}
+
+		loadBalancers := make([]*api.NetworkLoadBalancer, 0, len(records))
+		for _, record := range records {
+			loadBalancers = append(loadBalancers, record)
+		}
+
+		return response.SyncResponse(true, loadBalancers)
+	}
+
+	listenAddresses, err := d.State().DB.Cluster.GetNetworkLoadBalancerListenAddresses(n.ID(), memberSpecific)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("Failed loading network load balancers: %w", err))
+	}
+
+	loadBalancerURLs := make([]string, 0, len(listenAddresses))
+	for _, listenAddress := range listenAddresses {
+		u := api.NewURL().Path(version.APIVersion, "networks", n.Name(), "load-balancers", listenAddress)
+		loadBalancerURLs = append(loadBalancerURLs, u.String())
+	}
+
+	return response.SyncResponse(true, loadBalancerURLs)
+}
+
+// swagger:operation POST /1.0/networks/{networkName}/load-balancers network-load-balancers network_load_balancers_post
+//
+// Add a network address load balancer
```

Comment:
I'm not sure about the phrasing of `network address load balancer` as used in this file, can't it just be `network load balancer`?

As we handle traffic per port, `network address` feels odd.